constrained_linear_regression/__init__.py

```diff
@@ -78,6 +84,195 @@ def fit(self, X, y, min_coef=None, max_coef=None):
         return self
 
 
+class MultiConstrainedLinearRegression(ConstrainedLinearRegression):
+    """
+    This class implementation extends the ConstrainedLinearRegression class to handle multiple constraints.
+
+    Parameters
+    ----------
+    fit_intercept : bool, default=True
+        Whether or not to calculate the intercept for this model.
+
+    normalize : bool, default=False
+        This parameter is ignored when fit_intercept is set to False. 
+        If True, the regressors X will be normalized before regression by subtracting the mean and dividing by the l2-norm.
+
+    copy_X : bool, default=True
+        If True, X will be copied; else, it may be overwritten.
+
+    nonnegative : bool, default=False
+        When set to True, forces the coefficients to be positive.
+
+    ridge : float, default=0
+        When set to a float value, adds RIDGE regularization to the model, with the supplied alpha.
+
+    lasso : float, default=0
+        When set to a float value, adds LASSO regularization to the model, with the supplied alpha.
+
+    tol : float, default=1e-15
+        The tolerance for the optimization.
+
+    learning_rate : float, default=1.0
+        The learning rate for the optimization.
+
+    max_iter : int, default=10000
+        The maximum number of iterations for the optimization algorithm.
+
+    Methods
+    --------
+    fit(X, y, horizon, min_coef=None, max_coef=None, initial_beta=None):
```

Comment:
Can you please explain what the "horison" is and why a single model can be fit with different horisons? 
It is really not clear.

Reply:
I excluded horizon in fit, because `MultiConstrainedLinearRegression.global_horizon_count` can handle this.
horizon? [source](https://stats.stackexchange.com/questions/586244/what-does-it-mean-forecast-horizon-in-time-series-forecasting)

Point Forecast: In this interpretation, a forecast horizon of 30 days means predicting the value of the time series on the 30th day in the future. This approach is used when you only care about a specific point in time.

Multi-Step Forecast: In this interpretation, a forecast horizon of 30 days means predicting the values for all 30 days in the future. There are two main types of multi-step forecasting:

Direct Multi-step Forecast: Separate models are created for each step in the forecast horizon, e.g., one model for predicting the value one day ahead, another for two days ahead, and so on, up to 30 days ahead.
Iterative (Recursive) Multi-step Forecast: A single model is used to iteratively predict future values. The model predicts the value one step ahead, and then this predicted value is fed back into the model to predict the value two steps ahead, and so on.

In this case, it is for Multi-Step.